lib/internal/fs/promises.js

```diff
@@ -209,6 +215,22 @@ class FileHandle extends EventEmitterMixin(JSTransferable) {
     return this[kClosePromise];
   }
 
+  /**
+   * @typedef {import('../webstreams/readablestream').ReadableStream
+   * } ReadableStream
+   * @returns {ReadableStream}
+   */
+  readableWebStream() {
+    if (this[kFd] === -1)
+      throw new ERR_INVALID_STATE('The FileHandle is closed');
+    if (this[kClosePromise])
+      throw new ERR_INVALID_STATE('The FileHandle is closing');
+    if (this[kLocked])
+      throw new ERR_INVALID_STATE('The FileHandle is locked');
+    this[kLocked] = true;
+    return newReadableStreamFromStreamBase(this[kHandle]);
+  }
```

Comment:
Does this need a lock? Can't it just keep the position internally so that you might create multiple independent readers? Also could we have `start` & `end` as parameters?

Reply:
Also, I think this might need a `this[kRef]()` and unref on close otherwise the stream will fail with undefined behavior if the file handle is closed?